javascript-express/app.js

```diff
@@ -182,18 +186,59 @@ app.post('/delete_hook', urlencodedParser, function(req, res) {
   var subscriptionId = req.body.subscriptionId || '';
   if (subscriptionId.match(/^[0-9a-f-]+$/)) {
     rcsdk.platform()
-      .delete('/subscription/' + subscriptionId)
-      .then(function(response) {
-        console.log('DELETE THEN');
-        res.send('DELETE THEN: ' + subscriptionId);
-      })
-      .catch(function(e) {
-        console.log('DELETE CATCH');
-        console.log(e);
-        res.send('DELETE CATCH: ' + e);
-      }) 
+        .delete('/subscription/' + subscriptionId)
+        .then(function(response) {
+          console.log('DELETE THEN');
+          res.send('DELETE THEN: ' + subscriptionId);
+        })
+        .catch(function(e) {
+          console.log('DELETE CATCH');
+          console.log(e);
+          res.send('DELETE CATCH: ' + e);
+        })
   } else {
     console.log('DELETE ELSE');
     res.sendStatus(400);
   }
 });
+
+// To retrieve extensions
+app.post('/retreive_all_extensions', function(req, res) {
```

Comment:
This should be updated to handle accounts of any number of extensions.

As it is, the `POST /retreive_all_extensions` endpoint will [only retrieve up to 100 extensions which is the default `perPage` setting for the endpoint](https://developers.ringcentral.com/api-docs/latest/index.html#!#RefExtensionList.html).

To retrieve all extensions, we need to follow the `navigation` links when they exist. For example:

1. Set request `perPage` query parameter higher, e.g. to the max of 1000
2. Follow response `navigation.nextPage.uri` URIs when they exist

You can see an implementation of this in the Ruby SDK:

* [`RingCentralSdk::REST::Cache::Extensions.retrieve()`](https://github.com/grokify/ringcentral-sdk-ruby/blob/master/lib/ringcentral_sdk/rest/cache/extensions.rb#L24), lines [32](https://github.com/grokify/ringcentral-sdk-ruby/blob/master/lib/ringcentral_sdk/rest/cache/extensions.rb#L32) and [42](https://github.com/grokify/ringcentral-sdk-ruby/blob/master/lib/ringcentral_sdk/rest/cache/extensions.rb#L42)